contracts/jstz_fa_bridge.mligo

```diff
@@ -0,0 +1,127 @@
+#import "./common/entrypoints/ticketer-deposit.mligo" "TicketerDepositEntry"
+#import "./common/entrypoints/router-withdraw.mligo" "RouterWithdrawEntry"
+#import "./common/tokens/tokens.mligo" "Token"
+#import "./common/types/ticket.mligo" "Ticket"
+#import "./common/errors.mligo" "Errors"
+#import "./common/assertions.mligo" "Assertions"
+
+#include "./common/jstz_type.mligo"
+
+module Storage = struct
+  (* Stores the receiver and rollup address during a bridging process. 
+     It is set when deposit entrypoint is executed and cleared after 
+     the ticket is received from the ticketer. *)
+  type ongoing_deposit =
+    {
+    receiver : address;
+      (* Address that will receive tokens in Jstz *)
+    rollup : address
+      (* Address of the Jstz smart rollup contract *)
+    }
+
+  type t =
+    { 
+      token : Token.t;
+        (* Token which Ticketer accepts for minting tickets, immutable *)
+      ticketer : address;
+        (* Ticketer address, immutable *)
+      proxy : address option;
+        (* Proxy smart function address, immutable *)
+      ongoing_deposit : ongoing_deposit option;
+        (* Details of the on-going deposit *)
+      metadata : (string, bytes) big_map
+        (* Metadata of the contract (TZIP-016), immutable *)
+    }
+
+  let set_ongoing_deposit (ongoing_deposit: ongoing_deposit) (store : t) : t =
+    {store with ongoing_deposit = (Some ongoing_deposit)}
+
+  let clear_ongoing_deposit (store : t) : t = {store with ongoing_deposit = None}
+end
+
+module Jstz_fa_bridge = struct
+  module Tezos = Tezos.Next
+
+  type storage = Storage.t
+
+  type return = operation list * storage
+
+
+  type deposit_params =
+    { 
+      rollup : address;
+       (* Jstz rollup address *)
+      receiver : address; 
+       (* Address in Jstz that will receive tokens *)
+      amount : nat
+       (* Amount of tokens to bridge *)
+    }
+
+
+  (* [deposit params store] entrypoint is called when the user wants to bridge tokens
+      to Jstz.
+      
+     This entrypoint will transfer tokens from the user to the contract
+     and then call `Ticketer.deposit` entrypoint, which will mint a ticket
+     and send it back to this contract triggering the `default` entrypoint. *)
+  [@entry]
+  let deposit (params : deposit_params) (store : storage) : return =
+    let {
+      amount;
+      receiver;
+      rollup
+    } = params in
+    let () = Assertions.no_xtz_deposit () in
+    let token = store.token in
+    let ticketer = store.ticketer in
+    let sender = Tezos.get_sender () in
+    let self = Tezos.get_self_address () in
+    let token_transfer_op = Token.send_transfer token amount sender self in
+    let start_deposit_op = TicketerDepositEntry.send ticketer amount in
+    let approve_token_op = Token.send_approve token ticketer amount in
```

Comment:
nit: Might want to change the order of these in the code (to match their execution order)

Also what is the role of `Token.send_approve` -- the comment doesn't mention it?

Reply:
Reordered as suggested. 
`Token.send_approve` appoves the ticketer to spend `amount` tokens on behalf of the Bridge. It does this because when mint is called on the ticketer, the ticketer first transfers tokens from the sender (Bridge) to itself, then mints tickets and sends it back to the sender.